### ISSUE TITLE
Support Creating Tasker Action Plugin

### ISSUE THREAD

**Issue by k9withabone** [Jellepepe/flutter_tasker#1]
Any plans on adding support for creating a Tasker action plugin? For example, allowing a user to change an app setting from Tasker.

**Comment by Jellepepe** [comment]
The plugin feature was not originally required for the project I developed this plugin for, so it wasn't included.
From my understanding the tasker plugin system requires manifest declarations, so they would need to be added manually by developers, unless you imagine adding few premade actions?

The inclusion of some example actions could be possible. Perhaps a code generation module to generate the required code for such actions on the native side is feasible, but I do not have the time to work on these myself any time soon.

Feel free to do so and submit a pull request though!

**Comment by k9withabone** [comment]
As an update, I did figure out how to create a Tasker action plugin that works with Flutter. I've created an [example project](https://github.com/k9withabone/flutter_tasker_integration_example) that shows how to create an action plugin and a event plugin. I'm still trying to figure out how to do a state plugin.

For the action and event plugins, the Tasker runner is on a background thread so I can block it with a `Looper` while waiting for the Flutter/Dart code to finish on the main thread. Then, I can return any result that is communicated back for a output.

For the state plugin however, the runner is on the main thread already. If I try to block it waiting for the Flutter/Dart code to finish, I can't communicate back to the native side with the result since Flutter uses the main thread for communication. I am new to Android development so I am open to any ideas on how to work around this issue.

Also, I have no idea on how to go about turning my example into a code generation package. Any advice on that front would also be appreciated. I am using the [pigeon package](https://pub.dev/packages/pigeon) for Flutter to native communication in my example, so I'm thinking it could work similar to pigeon, potentially integrating pigeon itself into the package.
